tools/ddoc_xml_extractor.d

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env rdmd
+/*
+ * Extracts the newest articles from dlang.org's XML site and
+ * converts them to Ddoc macros.
+ *
+ * Copyright (C) 2018 by D Language Foundation
+ *
+ * Distributed under the Boost Software License, Version 1.0.
+ *    (See accompanying file LICENSE_1_0.txt or copy at
+ *          http://www.boost.org/LICENSE_1_0.txt)
+*/
+// Written in the D programming language.
+
+import std.format : format;
+import std.datetime;
+
+// crude representation of an article at the DBlog
+struct Article
+{
+    string title;
+    string link;
+    SysTime date;
+    string author;
+}
+
+auto toDdoc(const ref Article article, string suffix)
+{
+    import std.xml : encode;
+    string text;
+with(article)
+{
+    auto prettyMonth = ["January", "February", "March", "April", "May", "June", "July",
+    "August", "September", "October", "November", "December"][date.month-1];
```

Comment:
Woah, what happened to the indentation here

Reply:
-> https://github.com/dlang/dlang.org/pull/2305